packages/apps/project/src/api/project-task-api.js

```diff
@@ -111,3 +116,25 @@ export async function fetchProjectPriority() {
     page: 0,
   });
 }
+
+export async function fetchProjectTaskById({projecTaskId}) {
+  return createStandardFetch({
+    model: 'com.axelor.apps.project.db.ProjectTask',
+    id: projecTaskId,
+    fieldKey: 'project_projectTask',
+  });
+}
+
+export async function fetchTags({tagIds}) {
+  if (!Array.isArray(tagIds) || tagIds.length === 0) {
+    return {data: {data: [], total: 0}};
+  }
+
+  return createStandardSearch({
+    model: 'com.axelor.apps.project.db.ProjectTaskTag',
+    fieldKey: 'project_projectTaskTag',
+    page: 0,
+    numberElementsByPage: null,
+    criteria: createProjectTaskTagCriteria({tagIds}),
+  });
+}
```

Comment:
Instead of doing search function for tag list, add relatedFields in createStandardFetch of fetchProjectTaskById with : 

```
export async function fetchProjectTaskById({projecTaskId}) {
  return createStandardFetch({
    model: 'com.axelor.apps.project.db.ProjectTask',
    id: projecTaskId,
    fieldKey: 'project_projectTask',
    relatedFields: {
        projectTaskTagSet: ['name', 'colorSelect'],
    }
  });
}
```

And remove all the logic you put before (translations, fieldKey, useMemo with ids...)